File: pm/src/main/java/org/sqlbuilder/pm/SensekeyResolver.java
package org.sqlbuilder.pm;

import org.sqlbuilder.common.DeSerialize;
import org.sqlbuilder.common.Resolver;

import java.io.File;
import java.io.IOException;
import java.util.AbstractMap.SimpleEntry;

public class SensekeyResolver extends Resolver<String, SimpleEntry<Integer, Integer>>
{
	public SensekeyResolver(final String ser) throws IOException, ClassNotFoundException
	{
		super(DeSerialize.deserialize(new File(ser)));
	}
}

File: pm/src/main/java/org/sqlbuilder/pm/SensekeyResolver.kt
package org.sqlbuilder.pm

import org.sqlbuilder.common.DeSerialize.deserialize
import org.sqlbuilder.common.Resolver
import java.io.File

typealias PmSensekeyResolvable = String
typealias PmSensekeyResolved = Pair<Int, Int>

class SensekeyResolver(ser: String) : Resolver<PmSensekeyResolvable, PmSensekeyResolved>(deserialize<Map<PmSensekeyResolvable, PmSensekeyResolved>>(File(ser)))
